### PR TITLE
[Backport release-22.11] netdata: backport security patches contained in 1.37.0

### PR DESCRIPTION
**Pull request by HLudmann**
###### Description of changes
Applying security patches contained in the [1.37.0 release](https://github.com/netdata/netdata/releases/tag/v1.37.0).
PRs identified as fixing the vulnerabilites:
- https://github.com/netdata/netdata/pull/14063
- https://github.com/netdata/netdata/pull/14064

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/netdata/default.nix [hunk 1]
```diff
@@ -59,6 +59,12 @@ in stdenv.mkDerivation rec {
     # Avoid build-only inputs in closure leaked by configure command:
     #   https://github.com/NixOS/nixpkgs/issues/175693#issuecomment-1143344162
     ./skip-CONFIGURE_COMMAND.patch
+    
+    # Fix stream authentication: https://github.com/netdata/netdata/pull/14063
+    ./fix-stream-authentication.patch
+
+    # Fix alarm notification commands: https://github.com/netdata/netdata/pull/14064
+    ./fix-alarm-notification-commands.patch
```
Comment (winterqt) [inline pkgs/tools/system/netdata/default.nix:67]: Please use `fetchpatch` with the merged upstream commit patches here, if they apply cleanly.

**Review by winterqt — CHANGES_REQUESTED** [review]

**Comment by winterqt** [comment]
Squash + force push, please.

**Review by RaitoBezarius — APPROVED** [review]
LGTM, with this, we can ship an acceptable version.

File: pkgs/tools/system/netdata/default.nix [hunk 2]
```diff
@@ -59,6 +60,20 @@ in stdenv.mkDerivation rec {
     # Avoid build-only inputs in closure leaked by configure command:
     #   https://github.com/NixOS/nixpkgs/issues/175693#issuecomment-1143344162
     ./skip-CONFIGURE_COMMAND.patch
+
+    (fetchpatch {
+      # Fix stream authentication: https://github.com/netdata/netdata/pull/14063
+      name = "fix-stream-authentication";
```
Comment (winterqt) [inline pkgs/tools/system/netdata/default.nix:66]: Having `.patch` in the name was right. I don't like having the redundant title there, maybe you can think of something better to go beside the link that isn't just the title?

```suggestion
      # https://github.com/netdata/netdata/pull/14063
      name = "fix-stream-authentication.patch";
```

File: pkgs/tools/system/netdata/default.nix [hunk 3]
```diff
@@ -59,6 +60,20 @@ in stdenv.mkDerivation rec {
...
+      url = "https://github.com/netdata/netdata/commit/811028aea2f146cc0ac2bc403f7d692add400d63.patch";
+      hash = "sha256-Evp5EXcBrdqNwigofequALxOvmUr62hCC5YtXcvBhao=";
+    })
+
+    (fetchpatch {
+      # Fix alarm notification commands: https://github.com/netdata/netdata/pull/14064
+      name = "fix-alarm-notification-commands";
```
Comment (winterqt) [inline pkgs/tools/system/netdata/default.nix:73]:
```suggestion
      # https://github.com/netdata/netdata/pull/14064
      name = "fix-alarm-notification-commands.patch";
```

**Review by winterqt — CHANGES_REQUESTED** [review]
Just a nit or two.

File: pkgs/tools/system/netdata/default.nix (hunk 3 above)
Comment (HLudmann) [reply pkgs/tools/system/netdata/default.nix]: Found something for this one, what do you think?

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 230758` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>netdata</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
This does not build because the patches does not apply cleanly:
```
netdata> patching file streaming/rrdpush.c
netdata> Hunk #1 succeeded at 660 (offset -106 lines).
netdata> Hunk #2 succeeded at 694 (offset -106 lines).
netdata> Hunk #3 succeeded at 702 (offset -106 lines).
netdata> Hunk #4 succeeded at 724 (offset -106 lines).
netdata> patching file streaming/stream.conf
netdata> Hunk #1 succeeded at 115 (offset 4 lines).
netdata> Hunk #2 succeeded at 189 (offset -3 lines).
netdata> applying patch /nix/store/04z4ipdgdfd854ysaqcwmh3053k02ir4-fix-alarm-notification-commands.patch
netdata> patching file daemon/main.c
netdata> Hunk #1 succeeded at 1013 (offset -11 lines).
netdata> patching file daemon/unit_test.c
netdata> patching file daemon/unit_test.h
netdata> Hunk #1 FAILED at 3.
netdata> 1 out of 2 hunks FAILED -- saving rejects to file daemon/unit_test.h.rej
netdata> patching file health/health.c
netdata> Hunk #1 succeeded at 1111 with fuzz 2 (offset 1094 lines).
netdata> Hunk #2 FAILED at 374.
netdata> Hunk #3 FAILED at 445.
netdata> Hunk #4 FAILED at 514.
netdata> 3 out of 4 hunks FAILED -- saving rejects to file health/health.c.rej
netdata> patching file health/notifications/alarm-notify.sh.in
netdata> patching file libnetdata/inlined.h
netdata> Hunk #1 succeeded at 182 (offset 1 line).
netdata> patching file spawn/spawn.c
netdata> patching file spawn/spawn.h
```

**Comment by winterqt** [comment]
For future reference @HLudmann, please test your PRs before submitting.

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 230758` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netdata</li>
  </ul>
</details>

**Pull request merged**